src/structs/JGraph.jl

```diff
@@ -0,0 +1,208 @@
+            _global_layout(args...)
+        end, graph_animation)
+    elseif mode == :dynamic
+        return ((args...) -> _global_property_limits(args...), graph_animation)
+    end
+end
+
+function _global_property_limits(video, object, frames; kwargs...)
+    g = object.meta
+    for obj in g.ordering
+        if typeof(obj.meta) == GraphVertex
+            for (k, _) in obj.meta.property_style_map
+                val = g.get_node_attribute(k)
+                if typeof(val) == Real
+                    if !(k in keys(g.node_property_limits))
+                        g.node_property_limits[k] = (val, val)
+                    end
+                    g.node_property_limits[k] = (
+                        min(g.node_property_limits[k][1], val),
+                        max(g.node_property_limits[k][2], val),
+                    )
+                end
+            end
+        elseif typeof(obj.meta) == GraphEdge
+            for (k, _) in obj.meta.property_style_map
+                val = g.get_edge_attribute(k)
+                if typeof(val) == Real
+                    if !(k in keys(g.edge_property_limits))
+                        g.edge_property_limits[k] = (val, val)
+                    end
+                    g.edge_property_limits[k] = (
+                        min(g.edge_property_limits[k][1], val),
+                        max(g.edge_property_limits[k][2], val),
+                    )
+                end
+            end
+        end
+    end
+end
+
+function _global_layout(video, object, frame; kwargs...)
+    g = object.meta
+    if frame == first(get_frames(object))
+        layout_x = []
+        layout_y = []
+        if g.layout == :none
+            return
+        elseif g.layout == :spring
+            # Check due to some errors in calling spring_layout with an empty graph
+            if nv(g.adjacency_list.graph) > 0
+                layout_x, layout_y = spring_layout(g.adjacency_list.graph)
+            end
+        elseif g.layout == :spectral
+            # Check special property layout_weight is defined on edges and collect weights
+            edge_ordering_positions = edge_props(g.adjacency_list.graph)
+            weights = map(
+                (idx) -> get(g.ordering[idx].change_keywords, :layout_weight, 1),
+                edge_ordering_positions,
+            )
+            if nv(g.adjacency_list.graph) > 0
+                layout_x, layout_y = spectral_layout(g.adjacency_list, weights)
+            end
+        end
+        # Normalize coordinates between -0.5 and 0.5
+        coords = map((p) -> Point(p), collect(zip(layout_x, layout_y))) .- [(0.5, 0.5)]
+        # Scale to graph dimensions
+        coords = coords .* [(g.width, g.height)]
+        object.opts[:layout] = coords
+    end
+    # Now assign positions back to all nodes
+    for (idx, p) in enumerate(node_props(g.adjacency_list))
+        g.ordering[p].meta.opts[:position] = object.opts[:layout][idx]
```

Comment:
is this used to give the position to each node later in rendering?

Reply:
Yes.